### DISCUSSION TITLE
On windows platform, does not support the installation of virtual IP addresses

### DISCUSSION THREAD

**Discussion by xtsheng123** [strongswan/strongswan#727] [Q&A]
Dear all:
     I know from user documentation that strongswan has some limitations on windows. But I have to solve the problem. The questions are as follows:  strongSwan on Windows. Beside some other limitations, the kernel-iph networking backend currently does not support the installation of virtual IP addresses.Such addresses are usually assigned to road-warrior clients, making the strongSwan Windows port not usable as client for this particular scenario. 
     How can I avoid this problem in the above particular scenario?

**Comment by tobiasbrunner** [comment]
Similar to what we do on FreeBSD or macOS, via a TUN device that gets the virtual IP assigned. Not sure if this then works with the _kernel-iph_ plugin (i.e. the built-in IPsec stack) or you'd have to actually use [_kernel-libipsec_](https://wiki.strongswan.org/projects/strongswan/wiki/kernel-libipsec) (i.e. our userland IPsec implementation).
